examples/async-sqlite/index.js

```diff
@@ -0,0 +1,33 @@
+"use strict";
+
+const { promisify } = require("util");
+
+const { databaseNew, databaseClose, databaseInsert, databaseGetById } = require("./index.node");
+
+// Convert the DB methods from using callbacks to returning promises
+const databaseInsertAsync = promisify(databaseInsert);
+const databaseGetByIdAsync = promisify(databaseGetById);
+
+// Wrapper class for the boxed `Database` for idiomatic JavaScript usage
+class Database {
+    constructor() {
+        this.db = databaseNew();
```

Comment:
nit: we could simplify this with class properties:

```js
db = databaseNew();
```

Reply:
I would prefer to be explicit. Some developers may not be familiar because class fields are currently Stage 3 and not merged into the ECMAScript standard.

Reply:
+1, good callout